### PR TITLE
Bug1500720 5.6

### PR DESCRIPTION
**Pull request by laurynas-biveinis**
http://jenkins.percona.com/job/percona-server-5.6-param/1008/#showFailuresLink

### CODE REVIEW

**Comment by kbauskar** [comment]
if (!srv_read_only_mode
&& *success && *success
&& (srv_unix_file_flush_method == SRV_UNIX_O_DIRECT && (srv_unix_file_flush_method == SRV_UNIX_O_DIRECT

|| srv_unix_file_flush_method == SRV_UNIX_O_DIRECT_NO_FSYNC)) { + || srv_unix_file_flush_method == SRV_UNIX_O_DIRECT_NO_FSYNC
|| srv_unix_file_flush_method == SRV_UNIX_ALL_O_DIRECT)
&& access_type != OS_FILE_READ_WRITE_CACHED) {
Same logic exist in os_file_create_func.
As discussed over IM it would be good to create a macro for it and use it at both places.

**Comment by kbauskar** [comment]
LGTM

**Comment by laurynas-biveinis** [comment]
@kbauskar, I have updated the MP

File: storage/innobase/os/os0file.cc [hunk 1]
```diff
@@ -1308,6 +1308,33 @@ os_file_create_simple_func(
...
+static
+void
+os_file_set_nocache_if_needed(os_file_t file, const char* name,
+			      const char *mode_str, ulint type,
+			      ulint access_type)
+{
+	if (srv_read_only_mode || access_type == OS_FILE_READ_WRITE_CACHED)
+		return;
+
+	if (srv_unix_file_flush_method == SRV_UNIX_O_DIRECT
```
Comment (kbauskar) [inline storage/innobase/os/os0file.cc]: Condition needs to be:

if (srv_unix_file_flush_method == SRV_UNIX_ALL_O_DIRECT
||  (type != OS_LOG_FILE 
     && (srv_unix_file_flus_method= SRV_UNIX_O_DIRECT
            || srv_unix_file_flush_method == SRV_UNIX_O_DIRECT_NO_SYNC)) {
}

**Comment by laurynas-biveinis** [comment]
@kbauskar, I have updated the MP

**Comment by laurynas-biveinis** [comment]
Self-approving based on IM conversation

**Pull request merged**